Update cmake section in readme with required env-vars (#769)
compnerd/swift-win32: README.md

%CD%\build\bin\UICatalog.exe
```

Following environment variables should be set/adjusted before run the above commands (the shown paths assume that you use compiler from the official swift installers from [swift.org](https://www.swift.org/download/)).

- `SDKROOT`

  This should be the path to the Windows Platform SDK from the Swift library.

>   Command Prompt: `set SDKROOT=C:\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk;%PATH%`

Reviewer: This should be set automatically by the installer. If it is not, we should investigate what is going on. This is only needed during the build, not the run.

Author: After I restarted my machine it was the case. I assume that is the classic "Open from Windows Explorer" after the an installation, so the explorer don't got the new env-vars.

I would change it to "if it is not set" or should I remove it completely?

btw: Naming the system wide variable only `SDKROOT` sound like a high potential for a naming collision. `SWIFT_SDKROOT` would be a better choice in my opinion. I had a lot of tool which make similar decision and I always need to find workarounds for this type of issue.

Author: I've added the sentence that this should be the case if the installer was used.

Reviewer: Yeah, I think that is it - that starting from explorer would not get the new environment variables. I always just launch a new terminal instance, so I never noticed this and never thought to include that in the documentation.

Yeah, I agree with you about the naming. I would honestly prefer a nicer approach than using an environment variable. Note that the environment variable is _optional_. If you do not have it set, you need to explicitly pass the path along to the compiler for the `-sdk` parameter (which is implicitly set if `SDKROOT` is set).  However, the use of `SDKROOT` does mean that we have similar behaviour across platforms.

One final issue - the path is not guaranteed. It uses `%SystemDrive%` until 5.10.  With 5.10 the installer is now per-user and will be installed to `%LocalAppData%\Programs\Swift`, with the SDKs under `%LocalAppData%\Programs\Swift\Platforms`, toolchains under `%LocalAppData%\Programs\Swift\Toolchains`, and the runtime under `%LocalAppData%\Programs\Swift\Runtimes`.

Author: I added powershell commands as well as 5.10 variables

> - `PATH`

Reviewer: Windows spells this `Path` but it is case insensitive so it ends up being the same.

Author: Correct. This is something which confuse me for decades 😄

  The `PATH` environment variables needs to be extended that cmake can find the swift compiler. This is only required when this is not already the case. When you type `where swiftc` and see no path to swiftc the following command is required.

>   Command Prompt: `set PATH=C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin;%PATH%`

Reviewer: The tools path should also be setup by the installer. If this is not the case, we should investigate what is going on.

Author: Same as on the other part, after restart (a restart of the explorer process would do the same) it was set.

Author: I've added the sentence that this should be the case if the installer was used.

- Visual Studio compiler toolchain

>   The cmake commands use the `mt` tool which require a fully bootstrapped compiler environment. You can create a command prompt by click on `x64 Native Tools Command Prompt for VS2019` from the Visual Studio 2019 installation in the Windows Start Menu or call `C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Auxiliary\Build\vcvars64.bat` in the command prompt.

Reviewer: ```suggestion
The CMake commands use the `mt` tool which requires a Visual Studio compiler environment. You can create a command prompt by click on `x64 Native Tools Command Prompt for VS2019` from the Visual Studio 2019 installation in the Windows Start Menu or call `C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Auxiliary\Build\vcvars64.bat` in the command prompt.
```

Reviewer: I wonder if we should instruct people to use `vswhere` to find the latest environment. This would be better than assuming that VS2019 is installed (VS2022 could be installed), and more importantly, for the docker environment, the expectation is that we will have VS Build Tools, not VS2019.

Author: Mention `vswhere` could help, but it depends on the skill set of the user as well. The `x64 Native Tools Command Prompt for VS20xx` is super easy to start play around. `vswhere` could be a little more complicated but is the better automate able solution.

I added a small sentence about `vswhere`. Let me know if I should add more (I personally don't use `vswhere` directly, I use conan or cmake to bootstrap my environments).

- Visual Studio compiler toolchain

  The CMake commands use the `mt` tool which requires a Visual Studio compiler environment (Build Tools should be fine too).
  e.g. if you have Visual Studio 2019 installed you can create a command prompt by click on `x64 Native Tools Command Prompt for VS2019` from the Visual Studio 2019 installation in the Windows Start Menu or call `C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Auxiliary\Build\vcvars64.bat` in the command prompt.
  
  Higher versions of Visual Studio or the corresponding Build Tools should be fine too.

  Another way of activating a Visual Studio environment is the tool [vswhere](https://github.com/microsoft/vswhere).

- `SDKROOT`

  This should be the path to the Windows Platform SDK from the Swift library. This should be already set by the official swift installer. If not set the env var.

  Command Prompt: `set SDKROOT=C:\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk;%PATH%`

- `Path`

>   The `Path` environment variables needs to be extended that CMake can find the swift compiler. This is only required when this is not already the case. When you type `where swiftc` and see no path to swiftc the following command is required.

Reviewer: ```suggestion
  The `Path` environment variables needs to be extended to allow the Swift compiler to be found. This is only required when this is not already the case. When you type `where swiftc` and see no path to `swiftc` the following command is required.
```

Reviewer: `where swiftc` doesn't really work with PowerShell, so we would need to be more explicit. I think that perhaps we should just add a troubleshooting guide instead? We could put that on the Wiki so that we can make changes to this easier and grow it over time.

%CD%\build\bin\UICatalog.exe
```

> Following environment should be set/adjusted before run the above commands (the shown paths assume that you use compiler from the official swift installers from [swift.org](https://www.swift.org/download/)).

Reviewer: ```suggestion
Following environment variables should be set/adjusted before run the above commands (the shown paths assume that you use compiler from the official swift installers from [swift.org](https://www.swift.org/download/)).
```

- Visual Studio compiler toolchain

>   The CMake commands use the `mt` tool which requires a Visual Studio compiler environment (Build Tools should be fine too).

Reviewer: SPM also needs to use it, it just has to be done manually.

Author: Not quite sure what you mean here?
There is already the mention of the `mt` in the SPM section.
Should I rephrase here?

>   e.g. if you have Visual Studio 2019 installed you can create a command prompt by click on `x64 Native Tools Command Prompt for VS2019` from the Visual Studio 2019 installation in the Windows Start Menu or call `C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Auxiliary\Build\vcvars64.bat` in the command prompt.

Reviewer: Why Professional? I may be using Enterprise or Community. This is why I think that we should really just provide a VSWhere invocation instead.

Author: I removed the path to vsvars64.bat. So now it is the same as the section from SPM.
I agree that vswhere would be better but I never used it.

>   Higher versions of Visual Studio or the corresponding Build Tools should be fine too.

Reviewer: Higher is confusing here - because the enterprise SKU is considered higher. Perhaps "Newer"?

  Another way of activating a Visual Studio environment is the tool [vswhere](https://github.com/microsoft/vswhere).

- `SDKROOT`

  This should be the path to the Windows Platform SDK from the Swift library. This should be already set by the official swift installer. If not set the env var.

  \>= 5.10 `SDKROOT` needs to be `%LocalAppData%\Programs\Swift\Platforms\Windows.platform\Developer\SDKs\Windows.sdk`

>   < 5.10 `SDKROOT` needs to be `C:\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk`

Reviewer: Can we use `%SystemDrive%` instead of `C:` as the installer does?

  Examples for < 5.10:

  Command Prompt: `set SDKROOT=C:\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk`

>   PowerShell: `$env:SDKROOT="C:\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk"`

Reviewer: Similar

- `Path`

  The `Path` environment variables needs to be extended to allow the Swift compiler to be found.
  This is only required when this is not already the case.
  When you type `where swiftc` (`(Get-Command swiftc).Path` for PowerShell) and see no path to `swiftc` the following command is required.

>   \>= 5.10 the `Path` needs to extended with `%LocalAppData%\Programs\Swift\Runtimes\0.0.0\usr\bin;%LocalAppData%\Programs\Swift\Toolchains\0.0.0+Asserts\usr\bin`

Reviewer: The 0.0.0 will be actual release number of actual releases and 0.0.0 for snapshots or developer builds.

Author: I've added `5.9.0` as example
